File: voice-vault-frontend/types/note.ts
export const NOTE_STATUS = {
  UPLOADING: "UPLOADING",
  PROCESSING: "PROCESSING",
  READY: "READY",
  FAILED: "FAILED",
} as const;

export type NoteStatus = (typeof NOTE_STATUS)[keyof typeof NOTE_STATUS];

export interface Note {
  id: string;
  title: string;
  sourceFileName: string;
  category: string;
  status: NoteStatus;
  studyNotes: string | null;
  script: string | null;
  processingError: string | null;
  audioId: string | null;
  durationSeconds: number | null;
  createdAt: string;
  updatedAt: string;
}

File: voice-vault-backend/app/schemas/note.py
import uuid
from datetime import datetime

from app.schemas.base import CamelModel


class NoteOut(CamelModel):
    id: uuid.UUID
    title: str
    source_file_name: str
    category: str
    status: str
    study_notes: str | None
    script: str | None
    audio_id: uuid.UUID | None
    duration_seconds: int | None
    created_at: datetime
    updated_at: datetime
